src/apple/deps/update.rs

```diff
@@ -43,28 +50,51 @@ pub struct Outdated {
 }
 
 impl Outdated {
-    pub fn load() -> Result<Self, OutdatedError> {
+    pub fn load(gem_cache: &mut GemCache) -> Result<Self, OutdatedError> {
         #[derive(Deserialize)]
         struct Raw {
             formulae: Vec<Formula>,
         }
 
-        bossy::Command::impure_parse("brew outdated --json=v2")
+        let package_names = PACKAGES
+            .iter()
+            .map(|info| info.pkg_name)
+            .collect::<HashSet<_>>();
+
+        let gem_outdated = bossy::Command::impure_parse("gem outdated")
+            .run_and_wait_for_string()
+            .map_err(OutdatedError::CommandFailed)?;
+
+        let packages = bossy::Command::impure_parse("brew outdated --json=v2")
             .run_and_wait_for_output()
-            .map_err(Into::into)
+            .map_err(OutdatedError::CommandFailed)
             .and_then(|output| serde_json::from_slice(output.stdout()).map_err(Into::into))
-            .map(|Raw { formulae }| Self {
-                packages: formulae
+            .map(|Raw { formulae }| {
+                formulae
                     .into_iter()
-                    .filter(|formula| PACKAGES.contains(&formula.name.as_str()))
-                    .collect(),
-            })
+                    .filter(|formula| package_names.contains(&formula.name.as_str()))
```

Comment:
It was mostly to get rid of the O(N*M) scaling we have with `iter().find`. Also probably not super relevant since we only have like 4 Packages, but maybe in the future it could be noticeable if the person has a lot of installed gems.

Reply:
Gotcha. I prefer keeping things simple/naïve for now, and if we ever did need to speed this up, we'd be served better by using a `match` anyway.